server/bastionlab_common/src/common_conversions.rs

```diff
@@ -0,0 +1,316 @@
+                })?
+                .as_standard_layout(),
+        ),
+        DataType::Float64 => ndarray_to_tensor::<CowRepr<f64>, Ix2>(
+            &df.to_ndarray::<Float64Type>()
+                .map_err(|e| {
+                    Status::aborted(format!("Cound not convert DataFrame to ndarray: {}", e))
+                })?
+                .as_standard_layout(),
+        ),
+        DataType::Int64 => ndarray_to_tensor::<CowRepr<i64>, Ix2>(
+            &df.to_ndarray::<Int64Type>()
+                .map_err(|e| {
+                    Status::aborted(format!("Cound not convert DataFrame to ndarray: {}", e))
+                })?
+                .as_standard_layout(),
+        ),
+        DataType::Int32 => ndarray_to_tensor::<CowRepr<i32>, Ix2>(
+            &df.to_ndarray::<Int32Type>()
+                .map_err(|e| {
+                    Status::aborted(format!("Cound not convert DataFrame to ndarray: {}", e))
+                })?
+                .as_standard_layout(),
+        ),
+        DataType::Int16 => ndarray_to_tensor::<CowRepr<i16>, Ix2>(
+            &df.to_ndarray::<Int16Type>()
+                .map_err(|e| {
+                    Status::aborted(format!("Cound not convert DataFrame to ndarray: {}", e))
+                })?
+                .as_standard_layout(),
+        ),
+        DataType::Int8 => ndarray_to_tensor::<CowRepr<i8>, Ix2>(
+            &df.to_ndarray::<Int8Type>()
+                .map_err(|e| {
+                    Status::aborted(format!("Cound not convert DataFrame to ndarray: {}", e))
+                })?
+                .as_standard_layout(),
+        ),
+        _ => {
+            return Err(Status::aborted(format!("Unsupported datatype {}", dtype)));
+        }
+    }
+}
+
+pub fn tensor_to_series(name: &str, dtype: &DataType, tensor: Tensor) -> Result<Series, Status> {
+    Ok(match dtype {
+        DataType::Float32 => Series::from(tensor_to_chunked_array::<Float32Type>(&name, tensor)),
+        DataType::Float64 => Series::from(tensor_to_chunked_array::<Float64Type>(&name, tensor)),
+        DataType::Int64 => Series::from(tensor_to_chunked_array::<Int64Type>(&name, tensor)),
+        DataType::Int32 => Series::from(tensor_to_chunked_array::<Int32Type>(&name, tensor)),
+        DataType::Int16 => Series::from(tensor_to_chunked_array::<Int16Type>(&name, tensor)),
+        DataType::Int8 => Series::from(tensor_to_chunked_array::<Int8Type>(&name, tensor)),
+        d => {
+            return Err(Status::invalid_argument(format!(
+                "Unsuported data type in udf: {}",
+                d
+            )))
+        }
+    })
+}
+
+pub fn lazy_frame_from_logical_plan(plan: LogicalPlan) -> LazyFrame {
+    let mut ldf = LazyFrame::default();
+    ldf.logical_plan = plan;
+    ldf
+}
+
+pub fn to_status_error<T, E: Error>(input: Result<T, E>) -> Result<T, Status> {
+    input.map_err(|err| Status::aborted(err.to_string()))
+}
+
+pub fn load_udf(udf: String) -> Result<CModule, Status> {
```

Comment:
do we support udfs?

Reply:
Well, I think that's what `apply_udf` use. And if you want more info, ask Lucas.